ssas/service/public/api_test.go

```diff
@@ -906,3 +906,100 @@ func mintClientAssertion(issuer string, subject string, aud string, issuedAt int
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
+	handler := http.HandlerFunc(validateAndParseToken)
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusOK, rr.Result().StatusCode)
+	assert.Equal(s.T(), "application/json; charset=utf-8", rr.Result().Header.Get("Content-Type"))
+}
+
+func (s *APITestSuite) TestGetTokenInfoWithMissingToken() {
+	body := "{}"
+	req := httptest.NewRequest("POST", "/token_info", strings.NewReader(body))
+	rCtx := chi.NewRouteContext()
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rCtx))
+	handler := http.HandlerFunc(validateAndParseToken)
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusBadRequest, rr.Result().StatusCode)
+
+	var resMap map[string]string
+	err := json.NewDecoder(rr.Body).Decode(&resMap)
+	assert.NoError(s.T(), err)
+	assert.Equal(s.T(), "missing \"token\" field in body", resMap["error_description"])
+}
+
+func (s *APITestSuite) TestGetTokenInfoWithEmptyToken() {
+	body := `{"token":""}`
+	req := httptest.NewRequest("POST", "/token_info", strings.NewReader(body))
+	rCtx := chi.NewRouteContext()
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rCtx))
+	handler := http.HandlerFunc(validateAndParseToken)
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusBadRequest, rr.Result().StatusCode)
+
+	var resMap map[string]string
+	err := json.NewDecoder(rr.Body).Decode(&resMap)
+	assert.NoError(s.T(), err)
+	assert.Equal(s.T(), "missing \"token\" field in body", resMap["error_description"])
+}
+
+func (s *APITestSuite) TestGetTokenInfoWithCorruptToken() {
+	body := `{"token":"dafdasfdsfadfdasfdsafadsfadsf"}`
+
+	req := httptest.NewRequest("POST", "/token_info", strings.NewReader(body))
+	rCtx := chi.NewRouteContext()
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rCtx))
+	handler := http.HandlerFunc(validateAndParseToken)
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusOK, rr.Result().StatusCode)
+	assert.Contains(s.T(), rr.Body.String(), `{"valid":false}`)
+}
+
+func (s *APITestSuite) TestGetTokenInfoWithExpiredToken() {
+	_, access, err := s.MintTestAccessTokenWithDuration(time.Second * 1)
+	assert.NoError(s.T(), err)
+	time.Sleep(5 * time.Second)
+
+	body := fmt.Sprintf("{\"token\":\"%s\"}", access)
+	req := httptest.NewRequest("POST", "/token_info", strings.NewReader(body))
+	rCtx := chi.NewRouteContext()
+	req = req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rCtx))
+	handler := http.HandlerFunc(validateAndParseToken)
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(s.T(), http.StatusOK, rr.Result().StatusCode)
+	assert.Contains(s.T(), rr.Body.String(), `{"valid":false}`)
+}
+
+func (s *APITestSuite) MintTestAccessTokenWithDuration(duration time.Duration) (*jwt.Token, string, error) {
```

Comment:
It's worth trying to get a token in a different way than adding more token-creation code. This might be a good place to take advantage of (and update?) [this](https://github.com/CMSgov/bcda-ssas-app/blob/master/ssas/service/test_utils.go).

Reply:
We can grab an access token by calling the /token endpoint but we still need this method to create very short lived tokens.

We could modify the servers token duration, but that may introduce race conditions if tests are run in parallel.